src/AdjustSync.cpp

```diff
@@ -201,7 +201,10 @@ void AdjustSync::AutosyncOffset()
 		{
 			case AutosyncType_Song:
 			{
+				// Group offset must be preserved and applied to each step.
+				float groupOffset = GAMESTATE->m_pCurSong->m_SongTiming.m_fBeat0GroupOffsetInSeconds;
 				GAMESTATE->m_pCurSong->m_SongTiming.m_fBeat0OffsetInSeconds += mean;
+				GAMESTATE->m_pCurSong->m_SongTiming.m_fBeat0GroupOffsetInSeconds = groupOffset;
```

Comment:
I don't understand what's happening here:
We have:
```cpp
float groupOffset = GAMESTATE->m_pCurSong->m_SongTiming.m_fBeat0GroupOffsetInSeconds;
GAMESTATE->m_pCurSong->m_SongTiming.m_fBeat0GroupOffsetInSeconds = groupOffset;
```

you're just fetching the value and setting it back to the same thing??